compiler/rustc_driver/src/lib.rs

```diff
@@ -231,6 +231,10 @@ fn run_compiler(
         registry: diagnostics_registry(),
     };
 
+    if !tracing::dispatcher::has_been_set() {
+        init_rustc_env_logger_with_backtrace_option(&config.opts.unstable_opts.log_backtrace);
+    }
```

Comment:
This change is somewhat unfortunate for Miri, which was relying on `run_compiler` *not* initializing the logger... see https://github.com/rust-lang/miri/issues/2778. (That was a hack, but with the fact that logger filters cannot be changed at runtime, it's hard to avoid such hacks.)